File: app/src/main/java/adapter/RefuelDao.java
package adapter;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface RefuelDao {

    // Добавление Refuel2 в бд
    @Insert
    void insert(Refuel refuel);

    // Удаление Refuel2 из бд
    @Delete
    void delete(Refuel refuel);

    // Получение всех Refuel2 из бд
    @Query("SELECT * FROM Vehicle")
    List<Refuel> getAllRefuelList();

    // Получение всех Refuel2 из бд с условием
    @Query("SELECT * FROM Vehicle WHERE refuel_full_tank_flag LIKE :fullFlag")
    List<Refuel> getAllRefuelWithFullTankFlag(boolean fullFlag);

}

File: app/src/main/java/by/ewoks/powervehicle/common/dao/RefuelDao.kt
package by.ewoks.powervehicle.common.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import by.ewoks.powervehicle.common.entities.Refuel

@Dao
interface RefuelDao {

    @Insert
    fun insertRefuel(refuel: Refuel)

    @Query("SELECT * FROM Refuel")
    fun getAllRefuels(): LiveData<Refuel>
}